### PR TITLE
@types/dropzone Add missing upload augmentation to DropzoneFile.

### PR DESCRIPTION
**Pull request by macklin-10x**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: The extra parameters passed to File as Upload are partially documented here: https://www.dropzonejs.com/#tips
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

This PR adds the data structure that Dropzone adds to the `File` object that is handed to event handlers.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## dropzone ([<kbd>unpkg</kbd>](https://unpkg.com/browse/dropzone@latest/))
was missing the following properties:
1. version
1. optionsForElement
1. forElement
1. discover
1. blacklistedBrowsers

<details>
<summary>as well as these 5 other properties...</summary>
<p>dataURItoBlob, elementInside, getElement, getElements, isValidFile</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 837e83cadc8425c88082d26c0be94ff3c917c837
</p>

**Comment by macklin-10x** [comment]
It wasn't clear to me how the tests actually make assertions about expected behavior; I exercised the properties in a generic fashion but it could very well be that that execution path is never reached.

File: types/dropzone/index.d.ts [hunk 1]
```diff
@@ -26,13 +27,22 @@ declare namespace Dropzone {
         trgHeight?: number;
     }
 
+    export interface DropzoneFileUpload {
+        progress: number;
+        total: number;
+        bytesSent: number;
+        uuid: string;
+        totalChunkCount?: number;
```
Comment (PikachuEXE) [inline types/dropzone/index.d.ts:35]: I don't see `totalChunkCount` in doc
Can you explain more?

Comment (macklin-10x) [reply types/dropzone/index.d.ts:35]: The docs are not very complete; I don't think they were fully updated after the chunked uploads feature was added.  There are actually a few more fields in there.  The contents of `progress` doesn't seem to make sense, and `chunked: boolean` only appears when running in chunked mode.

Screen Shot 2020-05-20 at 9 19 31 AM

For a non-chunked upload, the fields are `bytesSent`, `filename`, `progress`, `total`, and `uuid`:

Screen Shot 2020-05-20 at 9 24 17 AM

**Review by PikachuEXE — APPROVED** [review]
I see
I can see `totalChunkCount` in source code
You can use references to source code next time

**Comment by macklin-10x** [comment]
How does this get merged?  It looks like the Travis build failed but for reasons unrelated to the code changes, and I can't figure out how to trigger it to run again.

**Comment by macklin-10x** [comment]
Ready to merge

**Pull request merged**